src/include/txn.i

```diff
@@ -1339,7 +1339,7 @@ __wt_txn_am_oldest(WT_SESSION_IMPL *session)
 	txn = &session->txn;
 	txn_global = &conn->txn_global;
 
-	if (txn->id == WT_TXN_NONE)
+	if (txn->id == WT_TXN_NONE || F_ISSET(txn, WT_TXN_PREPARE))
```

Comment:
We shouldn't roll back a prepared transaction here. We set `WT_SESSION_IGNORE_CACHE_SIZE` in a lot of places that deal with restoring from lookaside, but I saw cases where we got here with `WT_TXN_PREPARE` set.